### PR TITLE
Functional tests

### PR DESCRIPTION
**Pull request by matt-gadd**
This is an in the "spirit of" rather than absolute port of https://github.com/tastejs/todomvc/tree/master/tests

All page based actions are performed from the `Page` class, and in theory all the selectors should be kept in the `Selector` class. Some of the `Selector` functions are direct copies from tastejs.

The major differences between this and the tastejs version are that I have preferred to keep all assertions in the test file itself.

I have implemented 15 of the 30 tests to get us going.

The PR contains a few trivial fixes to pass some of the tests.

### CODE REVIEW

File: todo-mvc/tests/functional/main.ts [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+				.then(() => Promise.all([ page.getItem(0), page.getItem(1), page.getItem(2) ])
+				.then(([ item1, item2, item3 ]) => {
+					assert.equal(item1, TODO_ITEM_ONE);
+					assert.equal(item2, TODO_ITEM_TWO);
+					assert.equal(item3, TODO_ITEM_THREE);
+				}));
+		});
+
+		test.it('should trim text input', function () {
+			this.skip();
```
Comment (matt-gadd) [inline todo-mvc/tests/functional/main.ts]: skipped due to us currently failing.

File: todo-mvc/tests/functional/main.ts [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+	test.beforeEach(() => {
+		page = new Page(this.remote);
+		return page.init();
+	});
+
+	test.describe('When page is initially opened', function () {
+
+		test.it('should focus on the todo input field', function () {
+			this.skip();
```
Comment (matt-gadd) [inline todo-mvc/tests/functional/main.ts]: skipped due to us currently failing.

**Comment by agubler** [comment]
Do you think we should get this running CI as part of this PR? Or address that afterwards?

**Comment by matt-gadd** [comment]
@agubler beyond the scope of this PR given no one has added a `travis.yml` to even build it, and i'm not sure how the multi-folder set up would be.

**Comment by agubler** [comment]
Indeed there is not yet a `travis.yml` file, it was just a thought.

**Comment by matt-gadd** [comment]
@agubler we should probably have a separate issue for setting up our standard build for travis

**Comment by agubler** [comment]
@matt-gadd there is actually one already https://github.com/dojo/examples/issues/17

**Comment by matt-gadd** [comment]
@agubler awesome.

**Pull request merged**